iree_tests/conftest.py

```diff
@@ -220,12 +228,18 @@ def discover_test_cases(self):
                     continue
                 for test_case_json in test_cases_json["test_cases"]:
                     test_case_name = test_case_json["name"]
-                    have_all_files = self.check_for_remote_files(test_case_json)
+                    have_remote_files = self.check_for_remote_files(test_case_json)
+                    have_all_files = have_lfs_files and have_remote_files
+
+                    if not skip_missing and not have_all_files:
+                        raise FileNotFoundError(
+                            f"Missing files for test {self.path.parent.name}::{test_case_name}"
```

Comment:
lol this raises one exception per thread if using pytest-xdist... oh well. Seems fine...

```
pytest -k resnet50 -n auto --no-skip-tests-missing-files
======================================= test session starts ======================================= platform win32 -- Python 3.11.2, pytest-8.0.2, pluggy-1.4.0
rootdir: D:\dev\projects\SHARK-TestSuite\iree_tests
configfile: pytest.ini
plugins: dependency-0.6.0, retry-1.6.2, timeout-2.2.0, xdist-3.5.0
64 workers [0 items]    rror

============================================= ERRORS ==============================================
____________________ ERROR collecting pytorch/models/resnet50/resnet50.mlirbc _____________________
conftest.py:259: in collect
    test_cases = self.discover_test_cases()
conftest.py:235: in discover_test_cases
    raise FileNotFoundError(
E   FileNotFoundError: Missing files for test resnet50::real_weights
----------------------------------------- Captured stdout -----------------------------------------
Missing file 'inference_input.0.bin' for test resnet50::real_weights
Missing file 'inference_output.0.bin' for test resnet50::real_weights
Missing file 'real_weights.irpa' for test resnet50::real_weights
____________________ ERROR collecting pytorch/models/resnet50/resnet50.mlirbc _____________________
conftest.py:259: in collect
    test_cases = self.discover_test_cases()
conftest.py:235: in discover_test_cases
    raise FileNotFoundError(
E   FileNotFoundError: Missing files for test resnet50::real_weights
----------------------------------------- Captured stdout -----------------------------------------
Missing file 'inference_input.0.bin' for test resnet50::real_weights
Missing file 'inference_output.0.bin' for test resnet50::real_weights
Missing file 'real_weights.irpa' for test resnet50::real_weights
____________________ ERROR collecting pytorch/models/resnet50/resnet50.mlirbc _____________________
conftest.py:259: in collect
    test_cases = self.discover_test_cases()
conftest.py:235: in discover_test_cases
    raise FileNotFoundError(
E   FileNotFoundError: Missing files for test resnet50::real_weights

...
```